vcd/datasource_vcd_catalog.go

```diff
@@ -141,34 +148,87 @@ func datasourceVcdCatalog() *schema.Resource {
 	}
 }
 
+func getCatalogFromResource(catalogName string, d *schema.ResourceData, meta interface{}) (*govcd.AdminCatalog, error) {
+	vcdClient := meta.(*VCDClient)
+
+	orgName, err := vcdClient.GetOrgNameFromResource(d)
+	if err != nil {
+		return nil, fmt.Errorf("'org' property not supplied in the resource or in provider: %s", err)
+	}
+
+	tenantContext := govcd.TenantContext{}
+	if vcdClient.Client.IsSysAdmin {
+		org, err := vcdClient.GetAdminOrgByName(orgName)
+		if err != nil {
+			return nil, fmt.Errorf("[getCatalogFromResource] error retrieving org %s: %s", orgName, err)
+		}
+		tenantContext.OrgId = org.AdminOrg.ID
+		tenantContext.OrgName = orgName
+	}
+	catalogRecords, err := vcdClient.VCDClient.Client.QueryCatalogRecords(catalogName, tenantContext)
+	if err != nil {
+		return nil, fmt.Errorf("[getCatalogFromResource] error retrieving catalog records for catalog %s: %s", catalogName, err)
+	}
+	var catalogRecord *types.CatalogRecord
+	var orgNames []string
+	for _, cr := range catalogRecords {
+		orgNames = append(orgNames, cr.OrgName)
+		if cr.OrgName == orgName {
+			catalogRecord = cr
+			break
+		}
+	}
+	if catalogRecord == nil {
+		message := fmt.Sprintf("no records found for catalog '%s' from org '%s'", catalogName, orgName)
+		if len(orgNames) > 0 {
+			message = fmt.Sprintf("%s\nThere are catalogs with the same name from other orgs: %v", message, orgNames)
```

Comment:
This is a tiny suggestion to change "from org" into "in org", purely for clarity:

```suggestion
		message := fmt.Sprintf("no records found for catalog '%s' in org '%s'", catalogName, orgName)
		if len(orgNames) > 0 {
			message = fmt.Sprintf("%s\nThere are catalogs with the same name in other orgs: %v", message, orgNames)
```

Or maybe you had some specific architecture-driven reason for using the "from" which I'm missing?

Reply:
The first `from` should be changed to `in` as it conveys the meaning of belonging to the current org.
The second one should stay, as it is referring to catalog "shared from" other organizations. It may be clearer if I add the word "shared" before "from".

Reply:
On second thought, better use `in` everywhere, as we are not sure which Org was picked by the user